Use resolver classes from pyiron_snippets

pmrv opened the pull request:
Requires a version bump on snippets before merging.

pmrv committed:
Use ExecutableResolver

This also removes some backwards compatibility measures taken.
Previously Executable(module=None) was possible and we implicitly used
codename==module, but I think it's been broken all this while, because
e.g. `backwards_compatible_path_lst` doesn't add "/bin" to the paths to
search.  But it may also be from a time when our resources folders had a
different structure.

pmrv committed:
Use ResourceResolver in QueueAdapters

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

jan-janssen commented:
Waiting for https://github.com/pyiron/pyiron_base/pull/1533

pmrv commented:
> Waiting for #1533

Seems like we'll need Liam for it to make the conda release (since I don't have rights on the feedstock).  Since this PR is not critical, it doesn't need to be in the next release.

jan-janssen commented:
> > Waiting for #1533
> 
> Seems like we'll need Liam for it to make the conda release (since I don't have rights on the feedstock). Since this PR is not critical, it doesn't need to be in the next release.

It should be automated - just give it a bit of time 

EDIT: No Liam does not use the auto merge functionality. I requested access over a month ago but that pull request is still pending https://github.com/conda-forge/pyiron_snippets-feedstock/pull/2

jan-janssen committed:
Merge pull request #1545 from pyiron/main

Merge main

jan-janssen committed:
Update environment-old.yml

jan-janssen committed:
Update test_queue_adapter.py

jan-janssen committed:
Update unittests.yml

jan-janssen committed:
Update unittests_old.yml

jan-janssen commented on tests/unit/state/test_queue_adapter.py:
```diff
@@ -57,7 +57,7 @@ def tearDownClass(cls) -> None:
     def test_construction(self):
         self.assertEqual(
-            2,
+            3,
```
@pmrv Do you understand why the number of matches increased? I just want to prevent that the same files are read twice.

pmrv replied:
The new code finds these three files
```python
['/home/ponder/science/phd/dev/pyiron_base/testqueueadapters/res1/queues/queue.yaml',
 '/home/ponder/science/phd/dev/pyiron_base/testqueueadapters/res2/queues/queue.yaml', 
'/home/ponder/science/phd/dev/pyiron_base/testqueueadapters/res2/queues/clusters.yaml']
```
which are the ones that are created in the `setUpClass`.  It then creates a queue adapter for each of the files. But the way QueueAdapter works is that it wants to be passed the directory that the queue/cluster configuration, so the last two files give the same queue adapter.  It's an easy fix, but going forward do we even want to search for `clusters.yaml` anymore?

jan-janssen replied:
Yes, the `clusters.yaml` is used in the case when pyiron connects to multiple HPC clusters with different queuing systems. But I think it is fine, with the recent changes it only looks for directories and then `pysqa` is handling the rest.

pmrv committed:
Instantiate QueueAdapter only once per folder

In case both clusters.yaml and queues.yaml are present in a resource folder,
the old code created two (equivalent) queue adapters.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

jan-janssen reviewed (approved):
Looks good to me
